temperatures/crud.py

```diff
@@ -0,0 +1,17 @@
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from sqlalchemy import select
+from sqlalchemy.orm import selectinload
+
+from temperatures import models
+
+
+async def get_all_temperatures(db: AsyncSession, city_id: int | None = None):
+    query = select(models.Temperature).options(selectinload(models.Temperature.city))
+
+    if city_id is not None:
+        query = query.filter(models.Temperature.city.has(id=city_id))
```

Comment:
The `filter` method should be replaced with `where` when using SQLAlchemy's `select` construct. The `filter` method is typically used with the `Query` object, while `where` is used with `select` statements.